src/Mod/Sketcher/App/Sketch.cpp

```diff
@@ -2682,16 +2681,68 @@ int Sketch::addAngleAtPointConstraint(
     }
 
     int tag = -1;
-    if(e2c)
-        tag = Sketch::addPointOnObjectConstraint(geoId1, pos1, geoId2, driving);//increases ConstraintsCounter
+    // FIXME: Perform construction of any parameters where this method is called instead of here
+    if(e2c) {
+        if (Geoms[geoId2].type == BSpline) {
+            GCS::Point &p1 = Points[getPointId(geoId1, pos1)];
+            auto* partBsp = static_cast<GeomBSplineCurve*>(Geoms[geoId2].geo);
+            double uNear;
+            partBsp->closestParameter(Base::Vector3d(*p1.x, *p1.y, 0.0), uNear);
+            double* pointparam = new double(uNear);
+            Parameters.push_back(pointparam);
+            tag = addPointOnObjectConstraint(geoId1, pos1, geoId2, pointparam, driving); //increases ConstraintsCounter
+            --ConstraintsCounter;
+            GCSsys.addConstraintAngleViaPointAndParam(*crv2, *crv1, p, pointparam, angle, tag, driving);
+        }
+        else {
+            tag = Sketch::addPointOnObjectConstraint(geoId1, pos1, geoId2, driving);//increases ConstraintsCounter
+            GCSsys.addConstraintAngleViaPoint(*crv1, *crv2, p, angle, tag, driving);
+        }
+    }
     if (e2e){
         tag = ++ConstraintsCounter;
         GCSsys.addConstraintP2PCoincident(p, *p2, tag, driving);
+        GCSsys.addConstraintAngleViaPoint(*crv1, *crv2, p, angle, tag, driving);
     }
-    if(avp)
+    if(avp) {
         tag = ++ConstraintsCounter;
-
-    GCSsys.addConstraintAngleViaPoint(*crv1, *crv2, p, angle, tag, driving);
+        if (Geoms[geoId1].type == BSpline || Geoms[geoId2].type == BSpline) {
+            if (Geoms[geoId1].type == BSpline && Geoms[geoId2].type == BSpline) {
+                GCS::Point &p3 = Points[getPointId(geoId3, pos3)];
+                auto* partBsp = static_cast<GeomBSplineCurve*>(Geoms[geoId1].geo);
+                double uNear;
+                partBsp->closestParameter(Base::Vector3d(*p3.x, *p3.y, 0.0), uNear);
+                double* pointparam1 = new double(uNear);
+                Parameters.push_back(pointparam1);
+                addPointOnObjectConstraint(geoId3, pos3, geoId1, pointparam1, driving); //increases ConstraintsCounter
+                --ConstraintsCounter;
+                partBsp = static_cast<GeomBSplineCurve*>(Geoms[geoId2].geo);
+                partBsp->closestParameter(Base::Vector3d(*p3.x, *p3.y, 0.0), uNear);
+                double* pointparam2 = new double(uNear);
+                addPointOnObjectConstraint(geoId3, pos3, geoId2, pointparam2, driving); //increases ConstraintsCounter
+                Parameters.push_back(pointparam2);
+                --ConstraintsCounter;
+                GCSsys.addConstraintAngleViaPointAndTwoParams(*crv1, *crv2, p, pointparam1, pointparam2, angle, tag, driving);

```

Comment:
Note that the parameter here has to be the same for point-on-object and angle-via-point, so these constraints are best combined. This makes other point-on-object constraints partially redundant (unsure why they are not marked completely redundant).